File: vct-operation/platform_ios/VCTManager.m
//
//  VCTManager.m
//  operation
//
//  Created by Vincent on 15-1-26.
//
//

#import "VCTManager.h"
#import "VCTOC2CPP.h"

@interface VCTManager()
@property(nonatomic,retain) NSMutableDictionary *moduleDic;
@end

@implementation VCTManager

create_impl(VCTManager)

- (void)dealloc {
    self.moduleDic = nil;
    [super dealloc];
}

- (id)init {
    if (self = [super init]) {
        self.moduleDic = [[NSMutableDictionary alloc] init];
    }
    return self;
}

- (void)registerModule:(VCTModule *)module {
    if (!module) {
        NSLog(@"register module '%@' fail",module.moduleName);
        return;
    }
    [self.moduleDic setObject:module forKey:module.moduleName];
}

- (NSString *)requestWithModuleName:(NSString *)moduleName
                         MethodName:(NSString *)methodName
                              Param:(NSString *)param
                           Callback:(NSString *)callback {
    
    VCTModule *module = [self.moduleDic objectForKey:moduleName];
    if (!module) {
        NSLog(@"module '%@' not found",moduleName);
        return @"";
    }
    NSString *result = [module executeMethodWithName:methodName Param:param Callback:callback];
    return result;
}

- (void)response:(NSString *)param Callback:(NSString *)callback {
    [VCTOC2CPP response:param Callback:callback];
}

@end

File: vct-operation/platform_android/core/Manager/VCTManager.java
package Manager;
import java.util.HashMap;

import android.util.Log;

public class VCTManager {
	private static VCTManager mInstance = null;
	HashMap<String, VCTModule> mModuleMap = new HashMap<String, VCTModule>();   
	public static VCTManager getInstance(){
		if(mInstance == null){
			mInstance = new VCTManager();
		}
		return mInstance;
	}
	public void RegisterModule(VCTModule module){
		mModuleMap.put(module.GetModuleName(), module);
	}
	public String Request(final String moduleName,final String methodName,final String param,final String callback){
		if(mModuleMap.containsKey(moduleName))
		{
			VCTModule module = mModuleMap.get(moduleName);
			String result = module.Execute(methodName, param, callback);
			return result;
		}
		Log.i("AndroidLog","module " + moduleName + " not found");
		return "0";
	}
	private VCTManager(){
		
	};
}
